src/riak_kv_index_hashtree.erl

```diff
@@ -296,7 +305,7 @@ handle_call({delete, Items}, _From, State) ->
 handle_call(get_trees, _From, #state{trees=Trees}=State) ->
     {reply, Trees, State};
 
-handle_call({update_tree, Id}, From, State) ->
+handle_call({update_tree, Id, Callback}, From, State) ->
     lager:debug("Updating tree: (vnode)=~p (preflist)=~p", [State#state.index, Id]),
     apply_tree(Id,
                fun(Tree) ->
```

Comment:
after talking w/ @JeetKunDoug, it seems best to include a try/catch around maybe_callback below, but move this :deciduous_tree: fun into a separate named_fun.